packages/cli/src/services/check-parser/parser.ts

```diff
@@ -125,22 +148,35 @@ export class Parser {
     // Read the specific file if it has an extension
     if (path.extname(filePath).length) {
       const content = fs.readFileSync(filePath, { encoding: 'utf-8' })
-      return { filePath, content }
+      return [{ filePath, content }]
     } else {
-      if (preferedExtenstion === SupportedExtensions.JS) {
-        return Parser.tryReadFileExt(filePath, [SupportedExtensions.JS])
+      if (preferedExtenstion === FileExtensions.JS) {
+        return Parser.tryReadFileExt(filePath, JS_RESOLVE_ORDER)
       } else {
-        return Parser.tryReadFileExt(filePath, [SupportedExtensions.TS, SupportedExtensions.JS])
+        return Parser.tryReadFileExt(filePath, TS_RESOLVE_ORDER)
       }
     }
   }
 
-  static tryReadFileExt (filePath: string, exts: Array<SupportedExtensions>) {
+  static tryReadFileExt (filePath: string, exts: Array<FileExtensions>) {
     for (const extension of exts) {
       try {
+        const deps = []
         const fullPath = filePath + extension
         const content = fs.readFileSync(fullPath, { encoding: 'utf-8' })
-        return { filePath: fullPath, content }
+        deps.push({ filePath: fullPath, content })
+        if (extension === FileExtensions.PACKAGE) {
+          const { main } = JSON.parse(content)
+          if (!main || !main.length) {
+            // No main is defined. This means package.json doesn't have a specific entry
+            continue
+          }
+          const mainFile = path.join(filePath, main)
+          deps.push({
+            filePath: mainFile, content: fs.readFileSync(mainFile, { encoding: 'utf-8' }),
```

Comment:
Do we need to automatically import the `main` file from a `package.json`?

Reply:
Yeah. For example with the cli, it is `"main": "dist/index.js"`. If main doesn't exist, you go for `index.js` directly

Reply:
I'm still confused, though.  In a check, what's the advantage of importing `package.json` and having the `main` automatically imported as well?

Reply:
If we don't have the package.json, `import './module'` will point to `module/index.js`. If we have the package.json in the runtime, it should point to wherever the package.json specifies

Reply:
Ah got it. Thank you for clarifying.